ios/RNPurchases.m

```diff
@@ -121,10 +121,8 @@ - (dispatch_queue_t)methodQueue
     [RCCommonFunctionality restoreTransactionsWithCompletionBlock:[self getResponseCompletionBlockWithResolve:resolve reject:reject]];
 }
 
-RCT_REMAP_METHOD(syncPurchases,
-                 syncPurchasesWithResolve:(RCTPromiseResolveBlock)resolve
-                 reject:(RCTPromiseRejectBlock)reject) {
-    [RCCommonFunctionality syncPurchasesWithCompletionBlock:[self getResponseCompletionBlockWithResolve:resolve reject:reject]];
+RCT_EXPORT_METHOD(syncPurchases) {
```

Comment:
this isn't ideal, but it reflects the current versions of `purchases-ios` and `purchases-android`: 
`purchases-android` doesn't have a completion block for the method, while `purchases-ios` does. 
So as a compromise, this just updates the iOS side to not have a completion block. 
I think the completion block is a good idea, so we might wanna add it to the backlog and make an update for it when possible. @vegaro thoughts?

Reply:
we have a task in the backlog for adding a completion block to syncPurchases in Android. What if we invalidate purchaser info and then call `getPurchaserInfo` for now in the hybrids so we don't have to delete this?

Reply:
tbh I'd rather do this for now and then do a clean version when we tackle the backlog item, because we really shouldn't need to do an extra call to purchaserInfo if we fixed it on `purchases-android`, right?